sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkOperations.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.batch.ItemBulkOperation;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.Beta;
+import reactor.core.publisher.Flux;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Utility for creating bulk operations which can be executed by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)} .
+ *
+ * Also while creating these operation, if some options which are only for individual operation can be provided by passing
+ * a {@link BulkItemRequestOptions} while creating the bulk operation.
+ *
+ * See also {@link BulkProcessingOptions}.
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class BulkOperations {
+
+    /**
+     * Instantiate an operation for Creating item in Bulk execution.
+     *
+     * @param <T> The type of item to be created.
+     *
+     * @param item A JSON serializable object that must contain an id property.
+     * @param partitionKey the partition key for the operation
+     *
+     * @return the bulk operation.
+     */
+    public static <T> CosmosItemOperation newCreateItemOperation(T item, PartitionKey partitionKey) {
```

Comment:
"new" prefix sounds very odd actually... are there any other options to consider here? "insert" may be?

Reply:
Insert, create collides with operation name actually.

Reply:
How about "compose" or a similar wording? "newCreate..." and "newDelete.." just does not sound right... Plus "new" has a whole different meaning in Java, and in that case it's better if using a constructor method and not a static one; though I will guess that will add an explosion of new classes, for each operation type...

Reply:
to me as well "new" prefix seems odd. But I don't have any better suggestion. how about not having any prefix at all? (e.g., `newCreateItemOperation()` -> `createItemOperation()`

Reply:
But this might seem like BulkOperation is executing the operation rather than returning the operation to be executed by another function. So that's why we were thinking to have a "get" or some similar verb.

Reply:
I agree with @rakkuma here, without prefix, it might confuse the user. I am fine with both `getCreateItemOperation()` or `composeCreateItemOperation()`.

Reply:
Is this something the central SDK team should decide?

Reply:
the SDK team decided on the "get" as the prefix: e.g.: getCreateItemOperation()